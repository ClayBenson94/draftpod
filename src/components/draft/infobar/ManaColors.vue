
<template>

<table class="table table-sm mtgdraft-deck-colors">
  <tbody>
  <tr v-for="color in color_counts" :key="color.img">
    <td><img :src="color.img" width=18> {{ color.name }}</td>
    <td>{{ color.count }}</td>
  </tr>
  </tbody>
</table>

</template>

<script>

import * as filters from '../../../store/card-filters'

export default {
  name: 'ManaColors',

  props: {
     cards: {
      type: Array,
      required: true
    }
  },

  computed: {
    color_counts: function() {
      let counts = {
        W: {
          name: "Plains",
          img: "images/mana-white.svg",
          count: 0
        },
        B: {
          name: "Swamp",
          img: "images/mana-black.svg",
          count: 0
        },
        U: {
          name: "Island",
          img: "images/mana-blue.svg",
          count: 0
        },
        R: {
          name: "Mountain",
          img: "images/mana-red.svg",
          count: 0
        },
        G: {
          name: "Forest",
          img: "images/mana-green.svg",
          count: 0
        },
        C: {
          name: "Colorless",
          img: "images/mana-colorless.svg",
          count: 0
        },
      };
      for (let i=0; i<this.cards.length; i++) {
        let card = this.cards[i];
        if (filters.land(card))
          continue;
        let colors = card.colors;
        if (colors.length === 0)
          counts["C"].count++;
        else
          for (let c=0; c<colors.length; c++)
            counts[colors[c]].count++;
      }

      counts = Object.keys(counts).map(val => counts[val]);
      return counts.sort(function(a, b) {
        return b.count - a.count;
      });
    }
  },


}

</script>

<style>

.mtgdraft-deck-colors {
  font-size: 0.8em;
  font-weight: 500;
  margin-top: 3px;
}

.mtgdraft-deck-colors th,
.mtgdraft-deck-colors td {
  border-top: none;
}

.mtgdraft-deck-colors td img {
  margin-top: -2px;
  margin-right: 6px;
  margin-left: 4px;
}

</style>