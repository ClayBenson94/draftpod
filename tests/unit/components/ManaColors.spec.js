

import { mount } from '@vue/test-utils'
import ManaColors from '../../../src/components/draft/infobar/ManaColors.vue'

import cards from '../data/cards.json'

describe('ManaColors.vue', () => {

  test('displays all colors', () => {
    const wrapper = mount(ManaColors, {
      propsData: { cards }
    });
    const colorRows = wrapper.findAll('.deck-colors tbody>tr');
    expect(colorRows).toHaveLength(6);

    // do a snapshot test
    expect(wrapper.element).toMatchSnapshot();
  });

});




