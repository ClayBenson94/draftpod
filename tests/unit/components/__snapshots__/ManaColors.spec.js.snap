// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManaColors.vue displays all colors 1`] = `
<table
  class="table table-sm deck-colors"
>
  <tbody>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-green.svg"
          width="18"
        />
         Forest
      
      </td>
       
      <td
        align="right"
      >
        
        6
      
      </td>
       
      <td
        align="right"
      >
        
        50%
      
      </td>
    </tr>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-red.svg"
          width="18"
        />
         Mountain
      
      </td>
       
      <td
        align="right"
      >
        
        3
      
      </td>
       
      <td
        align="right"
      >
        
        25%
      
      </td>
    </tr>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-blue.svg"
          width="18"
        />
         Island
      
      </td>
       
      <td
        align="right"
      >
        
        2
      
      </td>
       
      <td
        align="right"
      >
        
        17%
      
      </td>
    </tr>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-black.svg"
          width="18"
        />
         Swamp
      
      </td>
       
      <td
        align="right"
      >
        
        1
      
      </td>
       
      <td
        align="right"
      >
        
        8%
      
      </td>
    </tr>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-white.svg"
          width="18"
        />
         Plains
      
      </td>
       
      <td
        align="right"
      >
        
        0
      
      </td>
       
      <td
        align="right"
      >
        
        0%
      
      </td>
    </tr>
    <tr>
      <td
        width="60%"
      >
        <img
          src="/images/mana-colorless.svg"
          width="18"
        />
         Colorless
      
      </td>
       
      <td
        align="right"
      >
        
        0
      
      </td>
       
      <td
        align="right"
      >
        
        0%
      
      </td>
    </tr>
  </tbody>
</table>
`;
